sdk/core/azure-core/test/ut/transport_adapter.cpp

```diff
@@ -15,7 +15,11 @@ namespace Azure { namespace Core { namespace Test {
     std::vector<std::unique_ptr<Azure::Core::Http::HttpPolicy>> p;
     std::shared_ptr<Azure::Core::Http::HttpTransport> transport
         = std::make_shared<Azure::Core::Http::CurlTransport>();
+    Azure::Core::Http::RetryOptions opt;
+    opt.RetryDelay = std::chrono::milliseconds(10);
```

Comment:
Isn't the default retry 4 seconds, why do we want just a 10 ms delay?

Also, since it is exponential back-off, and we have 3 max retries, it would be ~10, 20, 40 ms, and  then stop.

https://github.com/Azure/azure-sdk-for-cpp/blob/7c3257879870369560358eb35b664f7a75b5d269/sdk/core/azure-core/inc/azure/core/http/policy.hpp#L133-L148

Reply:
just don't want to wait 4 sec, or even a second for a test that is expected to last less than a second
All we want is to re-send the request for the unit test.

Reply:
I see, gotcha. How does retrying more frequently (but the same number of times) help fix the intermittent failure?

Reply:
No, we were not doing any retry.
Unit test is currently using only the transport policy directly.
This PR adds the Retry policy to pipeline so we can start retrying.

Retry policy has some HTTP status codes to retry by default.
This PR also makes possible to watch the ERROR number upon an error, so we can understand what's the code the server is returning and maybe add the code the retry policy